Index: includes/common.inc
===================================================================
RCS file: /cvs/drupal/drupal/includes/common.inc,v
retrieving revision 1.491
diff -u -r1.491 common.inc
--- includes/common.inc	28 Oct 2005 01:06:36 -0000	1.491
+++ includes/common.inc	21 Nov 2005 14:54:05 -0000
@@ -290,6 +290,8 @@
   drupal_set_header('HTTP/1.0 403 Forbidden');
   watchdog('access denied', t('%page denied access.', array('%page' => theme('placeholder', $_GET['q']))), WATCHDOG_WARNING, l(t('view'), $_GET['q']));
 
+  $_SESSION['destination'] = $_GET['q'];
+
   $path = drupal_get_normal_path(variable_get('site_403', ''));
   $status = MENU_NOT_FOUND;
   if ($path) {
